src/python/pants/backend/java/dependency_inference/rules_test.py

```diff
@@ -0,0 +1,438 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.java.compile.javac import rules as javac_rules
+from pants.backend.java.compile.javac_binary import rules as javac_binary_rules
+from pants.backend.java.dependency_inference.java_parser import rules as java_parser_rules
+from pants.backend.java.dependency_inference.java_parser_launcher import (
+    rules as java_parser_launcher_rules,
+)
+from pants.backend.java.dependency_inference.rules import InferJavaImportDependencies
+from pants.backend.java.dependency_inference.rules import rules as dep_inference_rules
+from pants.backend.java.target_types import (
+    JavaSourceField,
+    JavaSourcesGeneratorTarget,
+    JunitTestsGeneratorTarget,
+)
+from pants.backend.java.target_types import rules as java_target_rules
+from pants.backend.java.test.junit import rules as junit_rules
+from pants.backend.java.util_rules import rules as java_util_rules
+from pants.core.util_rules import config_files, source_files
+from pants.core.util_rules.external_tool import rules as external_tool_rules
+from pants.engine.addresses import Address, Addresses, UnparsedAddressInputs
+from pants.engine.target import (
+    Dependencies,
+    DependenciesRequest,
+    ExplicitlyProvidedDependencies,
+    InferredDependencies,
+    Targets,
+)
+from pants.jvm.resolve.coursier_fetch import rules as coursier_fetch_rules
+from pants.jvm.resolve.coursier_setup import rules as coursier_setup_rules
+from pants.jvm.util_rules import rules as util_rules
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *config_files.rules(),
+            *coursier_fetch_rules(),
+            *coursier_setup_rules(),
+            *dep_inference_rules(),
+            *external_tool_rules(),
+            *java_parser_launcher_rules(),
+            *java_parser_rules(),
+            *java_target_rules(),
+            *java_util_rules(),
+            *javac_binary_rules(),
+            *javac_rules(),
+            *junit_rules(),
+            *source_files.rules(),
+            *util_rules(),
+            QueryRule(Addresses, [DependenciesRequest]),
+            QueryRule(ExplicitlyProvidedDependencies, [DependenciesRequest]),
+            QueryRule(InferredDependencies, [InferJavaImportDependencies]),
+            QueryRule(Targets, [UnparsedAddressInputs]),
+        ],
+        target_types=[JavaSourcesGeneratorTarget, JunitTestsGeneratorTarget],
+        bootstrap_args=["--javac-jdk=system"],  # TODO(#12293): use a fixed JDK version.
+    )
+
+
+def test_infer_java_imports_same_target(rule_runner: RuleRunner) -> None:
```

Comment:
Use `@maybe_skip_jdk_test` on all tests until we get a better handle on the CI flakiness.

Reply:
Done